### PR TITLE
orb-slot-ctrl: integration tests and OrbSlotCtrl struct

### PR DESCRIPTION
**Pull request by vmenge**
added integration tests and test utils on `orb-slot-ctrl`
test utils are meant to be used by other crates when testing operations done through `orb-slot-ctrl`.
small refactoring done with the introduction of `EfiVarDb` to acquire `EfiVar`  from arbitrary paths.

tested and working on an Orb :+1:

### CODE REVIEW

File: CODEOWNERS
Comment (TheButlah) [inline CODEOWNERS:2]: I don't want to add individual codeowners to specific software components. Anyone should be able to review code in this repo. We do not need to further bottleneck our review process.

Comment (vmenge) [reply CODEOWNERS:2]: I'm okay with postponing this discussion for now, but I strongly feel a cross-team monorepo does not work well without a _CODEOWNERS_ file. IMO every crate / lib should have a team as clear owner as long as we're not bottlenecked by a single person.

File: update-verifier/src/lib.rs
Comment (oldgalileo) [inline update-verifier/src/lib.rs:25]: I know we already went over this, but I'll admit I'm still skeptical of dependency injection _even in application code_ as the answer to testability.

We looked a bit together, and I'll admit that alternatives besides doing something a bit more hack-y don't immediately jump out at me here. But I think in general testing should be _**in service of**_ application code.

Comment (oldgalileo) [reply update-verifier/src/lib.rs:25]: I thought more about this, and I think the issue I have with this testing is that it's testing `File::open` and `File::read`. That test, applied to the `Fixture` here, does not match the real-world scenario of reading and writing to the `efifvars`.

File: orb-slot-ctrl/src/main.rs
Comment (oldgalileo) [inline orb-slot-ctrl/src/main.rs:13]: Why do we move `program` into a new file?

File: orb-slot-ctrl/src/efivar/bootchain.rs
Comment (oldgalileo) [inline orb-slot-ctrl/src/efivar/bootchain.rs:11]: I know that you only changed their visibility, but it would be awesome to get a page reference to this in the EFI spec or wherever these values came from. Magic values with no documentation = sad face.

File: orb-slot-ctrl/src/main.rs
Comment (vmenge) [reply orb-slot-ctrl/src/main.rs:13]: We can't import things from `main.rs`  on integration tests :(

**Review by oldgalileo — APPROVED** [review]
@vmenge and I talked in person.

I stand by what I said before, that testing `fs::write` is **not** the aim of these tests. There's a point here that we do want to enable verification of the state after a series of operations (i.e. we have the `UpdateDone` rootfs status at the beginning of an integration-tested update-verifier run, and then we validate that it's `Normal` at the end). Remains to be seen how to appropriately implement that

File: orb-slot-ctrl/src/efivar/bootchain.rs
Comment (vmenge) [reply orb-slot-ctrl/src/efivar/bootchain.rs:11]: agreed! i'll chase down where these come from and document them later :+1:

File: orb-slot-ctrl/src/test_utils.rs
Comment (TheButlah) [inline orb-slot-ctrl/src/test_utils.rs:36]: Explain these magic values.

File: orb-slot-ctrl/tests/orb_slot_ctrl.rs
Comment (TheButlah) [inline orb-slot-ctrl/tests/orb_slot_ctrl.rs:12]: Question: Do you have an opinion on whether test functions should start their names with `test_` or not? In some cases I think it could disambiguate cases where we forgot to annotate with #[test]. But IDK.

File: orb-slot-ctrl/tests/orb_slot_ctrl.rs
Comment (TheButlah) [inline orb-slot-ctrl/tests/orb_slot_ctrl.rs:21]: Why would it default to the next slot being B? I would assume A is the current, and the next slot remains A.
Is this a product of the mystery writes in the Fixture code?

File: update-verifier/src/main.rs
Comment (TheButlah) [inline update-verifier/src/main.rs:30]: Why do it this way?


Also, main should be using `color_eyre::Result` along with a `color_eyre::install()?`

**Review by TheButlah — CHANGES_REQUESTED** [review]
Overall looks great! Thanks @vmenge :3

Just a few small changes and we will be good.

File: update-verifier/src/main.rs
Comment (vmenge) [reply update-verifier/src/main.rs:30]: > Why do it this way?
No need to manually call process::exit like was being done befoe

> Also, main should be using color_eyre::Result along with a color_eyre::install()?
Sure

File: orb-slot-ctrl/src/test_utils.rs
Comment (vmenge) [reply orb-slot-ctrl/src/test_utils.rs:36]: I got them all from the unit tests from `rootfs.rs`  and `bootchain.rs`, and commits were done by you and Mark Eder, so you probably can explain them better than me :sweat_smile:

File: orb-slot-ctrl/tests/orb_slot_ctrl.rs
Comment (vmenge) [reply orb-slot-ctrl/tests/orb_slot_ctrl.rs:12]: Nah, I don't think its necessary... starting tests with `it_` is a convention in some places as well, but I don't really think it matters since we have to annotate with `#[test]`.

File: orb-slot-ctrl/Cargo.toml
Comment (vmenge) [reply orb-slot-ctrl/Cargo.toml:32]: It's not due to tests though, it's due to the implementation which uses `libc` (you do have an upcoming change in the other PR). So the bin for it wouldn't even run in macOS.

I'll leave this as is for now as you can remove it once you've merged the removal of `libc`. I am not a fan of annotating every single test we have because the macOS pipeline can't run it tbh :/

**Pull request merged**
